core/src/main/scala/tech/beshu/ror/configuration/StartupConfig.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.configuration
+
+import tech.beshu.ror.accesscontrol.blocks.variables.startup.StartupResolvableVariableCreator
+import tech.beshu.ror.accesscontrol.blocks.variables.transformation.TransformationCompiler
+import tech.beshu.ror.providers.{EnvVarsProvider, JvmPropertiesProvider, OsEnvVarsProvider, PropertiesProvider}
+
+final case class StartupConfig(envVarsProvider: EnvVarsProvider,
+                               propertiesProvider: PropertiesProvider,
+                               variableCreator: StartupResolvableVariableCreator)
+
+object StartupConfig {
+  val default: StartupConfig = StartupConfig(
+    envVarsProvider = OsEnvVarsProvider,
+    propertiesProvider = JvmPropertiesProvider,
+    variableCreator = new StartupResolvableVariableCreator(TransformationCompiler.withoutAliases)
```

Comment:
I'm not sure if we should pass here the compiler or the variables creator. AFAIU we want to have the same compiler in all creator instances we use. Am I right? So maybe we can have several instances of the creator but we have to be sure that they use the same compiler.